### PR TITLE
Add fp8 quantization for conv and linear layers

### PR DESCRIPTION
**Pull request by nithinsubbiah**
This is a follow-up to (https://github.com/nod-ai/SHARK-Platform/pull/202) which injects fp8 linear and convolution kernels in punet.

### CODE REVIEW

File: sharktank/sharktank/kernels/templates/batch_matmul_transpose_b.mlir [hunk 1]
```diff
@@ -4,6 +4,7 @@
 // See https://llvm.org/LICENSE.txt for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+// !cst_zero = {{cst_zero}}
```
Comment (rsuderman) [inline sharktank/sharktank/kernels/templates/batch_matmul_transpose_b.mlir:7]: Delete comment out code

File: sharktank/sharktank/kernels/templates/batch_matmul_transpose_b.mlir [hunk 2]
```diff
@@ -15,7 +16,8 @@ module {
 util.func private @sharktank_batch_matmul_transpose_b_{{spec_sig}}(
     %a: !a_tensor_type, %b: !b_tensor_type)
     -> !c_tensor_type {
-  %zero = arith.constant 0: !dtype
+  // %zero = arith.constant !cst_zero: !dtype
```
Comment (rsuderman) [inline sharktank/sharktank/kernels/templates/batch_matmul_transpose_b.mlir:19]: More commented out code

File: sharktank/sharktank/ops/qconv_impls.py [hunk 3]
```diff
@@ -85,7 +89,10 @@ def qconv2d_tensor_scaled_integer(
 
     # Alias components (d=scale, qs=quantized samples, m=offset).
     if accum_dtype is None:
-        accum_dtype = torch.int32
+        if weight_layout.qs.dtype.is_floating_point:
```
Comment (rsuderman) [inline sharktank/sharktank/ops/qconv_impls.py:92]: I usually just do
```
accum_dtype = torch.int32
if weight_layout.qs.dtype.is_floating_point:
  accum_dtype = torch.float32
  ````

**Review by rsuderman — APPROVED** [review]

File: sharktank/sharktank/kernels/batch_matmul_transpose_b.py [hunk 4]
```diff
@@ -80,7 +80,7 @@ def generate(self, ksel: KernelSelection, kb: KernelBuilder):
         spec_sig = f"L{a_ident}_R{b_ident}"
         template_file = "batch_matmul_transpose_b.mlir"
         target_function_name = f"sharktank_batch_matmul_transpose_b_{spec_sig}"
-
+        cst_zero = "0." if "f" in str(accum_type) else "0"
```
Comment (stellaraccident) [inline sharktank/sharktank/kernels/batch_matmul_transpose_b.py:83]: Be specific. Don't just use stringy char-in-str.

In this case, I think you can say something like:

```
cst_zero = "0" if IntegerType.isa(accum_type) else "0."
```

**Pull request merged**